FDApy/representation/functional_data.py

```diff
@@ -0,0 +1,527 @@
+#!/usr/bin/python3.7
+# -*-coding:utf8 -*
+
+"""
+Module for the definition of FunctionalData types.
+
+This modules is used to defined different types of functional data. The
+different types are: Univariate Functional Data, Irregular Functional data and
+Multivariate Functional Data.
+"""
+import numpy as np
+
+from abc import ABC, abstractmethod
+
+from ..misc.utils import get_axis_dimension, get_dict_dimension, get_obs_shape
+
+
+###############################################################################
+# Checkers for parameters
+def _check_len(argv1, argv2):
+    """Raise an arror if `argv1` and `argv2` do not have the same length."""
+    if len(argv1) != len(argv2):
+        raise ValueError(f"{type(argv1).__name__} and {type(argv2).__name__}"
+                         " must have the same length.")
+    return None
+
+
+def _check_dict_dimension(argv1, argv2, axis=0):
+    """Raise an error in case of dimension conflicts along the `axis`.
+
+    An error is raised when elements of `argv1` and `argv2`, which are assumed
+    to be dictionary, do not have the length along the `axis`. Elements are
+    assumed to be numpy.ndarray.
+    """
+    has_len = [i.shape[0] == get_axis_dimension(j, axis)
+               for i, j in zip(argv1.values(), argv2.values())]
+    if not np.all(has_len):
+        raise ValueError(f"Dimension are not the same.")
+    return None
+
+
+def _check_dict_array(argv_dict, argv_array):
+    """Raise an error in case of dimension conflicts between the arguments.
```

Comment:
Codacy found an issue: [Multi-line docstring closing quotes should be on a separate line (D209)](https://app.codacy.com/manual/StevenGolovkine/FDApy/pullRequest?prid=5929506)